Automattic/pocket-casts-android
# Move Gradle configuration to build.gradle.kts

## Description

This PR is in response to https://github.com/Automattic/pocket-casts-android/pull/2644#discussion_r1719145599. This change is helpful for cleaning up Gradlea bit in any case but I plan on leaving `debugProd` only for apps in a follow-up PR. Key changes:

* Move configuration from Groovy `build.gradle` to Kotlin `build.gradle.kts`.
* Standardize minor things like R8 configuration across different projects.
* I did not move `dependnecies` block. It is a completely different can of worms and I don't want to deal with it right now. See: p1721250963767669-slack-C05S3A24L4T
* There are still some quirks like `minSdkVersionWear` but I didn't want to tackle it in this PR.

## Testing Instructions

1. Build different variants of `:app`, `:wear`, and `:automotive` and verify their signatures. See: https://github.com/Automattic/pocket-casts-android/pull/2051
2. Smoke test key features of the release app variants and check if nothing broke.

## Checklist
 
#### I have tested any UI changes...
